### DISCUSSION TITLE
How do I use cmake port scripts in my project?

### DISCUSSION THREAD

**Discussion by fromasmtodisasm** [microsoft/vcpkg#22784] [Q&A]
I want to write a cmake utility and form a port from it, I barely managed to understand how cmake-port does it, for some reason there is no documentation about such files as "vcpkg-port-config.cmake" anywhere, there is no information available in the documentation. But having this file, my cmake library can only be used in other portfile.cmake! And I want to specify this library in vcpkg.json as dependencies and use cmake functions from it. How can I achieve this?

**Comment by fromasmtodisasm** [comment]
I have an engine from which I made a port. It worked out well, I can install it via vcpkg. But it turned out that the dependent project requires cmake scripts that the engine itself uses. I need to somehow distribute these setup scripts along with the engine or as a separate package so that the consumer (my game) can be configured correctly

**Comment by fromasmtodisasm** [comment]
The problem is that earlier both the engine and the game lived in the same repository, after trying to move the engine to a separate repository, everything broke

**Comment by autoantwort** [comment]
I think it is not meant for end users, because of that there is no documentation. But it works as follows:  
1. Ports that want to distribute cmake scripts write and install a `vcpkg-port-config.cmake` file that includes the other cmake files.
2. vcpkg detects this files and pass them in the list `VCPKG_PORT_CONFIGS` to cmake: https://github.com/microsoft/vcpkg-tool/blob/main/src/vcpkg/build.cpp#L791-L805
3. In `ports.cmake` the files are included: https://github.com/microsoft/vcpkg/blob/master/scripts/ports.cmake#L130-L134

You could manually glob for these files in the vcpkg installed dir in your project and include these files yourself
